bin/generate-zbm

```diff
@@ -115,24 +121,44 @@ if ( ( defined $config{Global}{BootMountPoint} ) and ( length $config{Global}{Bo
 my $dir     = File::Temp->newdir();
 my $tempdir = $dir->dirname;
 
-# Set our kernel from the CLI, or pick the latest available in /boot
-if ( ( defined $runConf{kernel} ) and ( length $runConf{kernel} ) ) {
-  unless ( -f $runConf{kernel} ) {
-    printf "The provided kernel %s was not found, unable to continue", $runConf{kernel};
+if ( ! nonempty $runConf{kernel} ) {
+  # Try to determine a kernel file when one was not provided
+  if ( ! nonempty $runConf{kernel_version} ) {
+    $runConf{kernel} = latestKernel;
+  } else {
+    $runConf{kernel} = versionedKernel $runConf{kernel_version};
+  }
+
+  # Make sure a kernel was found
+  unless ( nonempty $runConf{kernel} ) {
+    print "Unable to choose a kernel file, cannot continue\n";
```

Comment:
Should this have an exit value of 1, since it's a failure?

Reply:
That's probably a good idea. I added a few other nonzero exit codes for error conditions for consistency.